include/picongpu/particles/particleToGrid/derivedAttributes/EnergyDensityCutoff.hpp

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2013-2017 Axel Huebl, Rene Widera, Heiko Burau, Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/particles/particleToGrid/derivedAttributes/EnergyDensityCutoff.def"
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/algorithms/KinEnergy.hpp"
+
+namespace picongpu
+{
+namespace particleToGrid
+{
+namespace derivedAttributes
+{
+
+    template< class T_ParamClass >
+    template< class T_Particle >
+    DINLINE float_X
+    EnergyDensityCutoff< T_ParamClass >::operator()( T_Particle& particle ) const
+    {
+        using ParamClass =  T_ParamClass;
+
+        /* read existing attributes */
+        const float_X weighting = particle[weighting_];
```

Comment:
- please add spaces `particle[ weighting_ ]`
- move const to the right of the type `float_X const weighting = ...` (please check also the two lines after this line)